File: data_structure/LeeJuYoung/programmers_stack_queue_2.py
def solution(progresses, speeds):
    require = []
    answer = []
    for i in range(len(speeds)):
        left_date = ceil((100-progresses[i]) / speeds[i])
        if require != [] and require[0] < left_date:
            answer.append(len(require))
            require = []
        require.append(left_date)
    answer.append(len(require))
    return answer

def ceil(n : float):
    if n // 1 == n:
        return int(n)
    else:
        return int(n//1 + 1)

File: data_structure/KimJunYoung/stackQueue/ProgrammersStackQueue2.java
package data_structure.KimJunYoung.stackQueue;

import java.util.*;

public class ProgrammersStackQueue2 {
    public static void main(String[] args) {
        ProgrammersStackQueue2 solved  = new ProgrammersStackQueue2();
        int[] test1Progresses = {95, 0};
        int[] test2Speeds = {4, 50};
        System.out.println(Arrays.toString(solved.solution(test1Progresses, test2Speeds))); // [2]
    }
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        for (int i = 0; i < speeds.length; i++) {
            int progress = progresses[i];
            int speed = speeds[i];
            int daysToDeploy = (int) Math.ceil((100 - progress) / (double) speed);
            queue.offer(daysToDeploy);
        }
        List<Integer> answer = new ArrayList<>();
        while(!queue.isEmpty()) {
            int daysToDeploy = queue.poll();
            int count = 1;
            while(!queue.isEmpty()) {
                if (queue.peek() > daysToDeploy) {
                    break;
                }
                queue.poll();
                count++;
            }
            answer.add(count);
        }
        return answer.stream().mapToInt(x -> x).toArray();
    }
}
